tests/testsuite.src/syn_definition.at

```diff
@@ -953,7 +960,11 @@ AT_DATA([prog.cob], [
            STOP RUN.
 ])
 
-AT_CHECK([$COMPILE_ONLY prog.cob], [1], [],
+AT_CHECK([$COMPILE_ONLY prog.cob], [0], [],
+[prog.cob: in section 'L':
+prog.cob:6: warning: non-unique section or paragraph name used
+])
+AT_CHECK([$COMPILE_ONLY -fnon-unique-procedure-names=error prog.cob], [1], [],
 [prog.cob: in section 'L':
 prog.cob:6: error: redefinition of 'L'
 prog.cob:5: note: 'L' previously defined here
```

Comment:
the FIXME below can be removed

Reply:
Is everything of `https://github.com/OCamlPro/gnucobol/pull/31#issuecomment-1137626598` already covered? If not then please add it as new testcase compiling once without std (expected warning with zero status) and with `-std-cobol2002` (expecting error -> status 1)

Reply:
I've added the new test; for now it does not pass and reports an ambiguity. We are currently checking whether this one should indeed pass on GCOS.

On that note, the FR links to a former implementation are all broken. Is there another place where that previous attempt can be investigated?

Reply:
Ah yes indeed; somehow I had forgotten that was a former name for gnucobol.

Anyways, as far as our tests go, GCOS appears to subject uses of identical names for sections/paragraphs and data to syntactic scoping rules (as mentioned by David in the FR). This is the behavior of the current implementation.
For this reason we should expect the test cases imported from the FR to fail.

Reply:
If there's a section and a paragraph defined and you have `PERFORM same-name` in GCOS that is auto-scoped?
So if you do it from `main section` it will call the section with that name, but if you call it from `other-section` that has the `same-name` paragraph defined it uses that?
I'm a bit skeptic.

Reply:
> If there's a section and a paragraph defined and you have `PERFORM same-name` in GCOS that is auto-scoped?
> So if you do it from `main section` it will call the section with that name, but if you call it from `other-section` that has the `same-name` paragraph defined it uses that?
> I'm a bit skeptic.

The part "what does GCOS do if there is both a section and a paragraph name with the same word" is the most important one not answered (or - totally possible: answered but I've overlooked that).

Would it be useful to go with a lesser version _first_ that matches the old FR? This would be `procedure-shares-data-name` or similar.